[Qu Zhetao] iP

quzhetao01 opened the pull request:
**Duke Task Bot**
This Task Bot serves to keep track of your tasks! It is
- Text-based
- Easy to Use
- Self-explanatory
AND **FREE!!** 😉

How to Setup:
1. Download it from [here](https://github.com/quzhetao01/ip/tree/master)
2. Run the program using `java -jar Duke.jar`
3. Add tasks 
4. Enjoy using the program and tracking your tasks!

Features:
- [ ] Managing ToDos
- [ ] Managing Deadlines
- [ ] Managing Events
- [ ] Reminders (coming soon)

Side note:
You can control where you want your tasks data to be stored on your device by changing details in the Duke constructor. Change the directory within the Storage constructor to your preferred destination. (It is set to "./tasks.txt" currently)
The Duke constructor is show below.
```
public Duke() {
        this.ui = new Ui();
        this.storage = new Storage("./tasks.txt");
        try {
            this.tasks = new FunnyList(storage.load());
        } catch (DukeException e) {
            ui.showLoadingError();
            this.tasks = new FunnyList();
        }
    }
```

Plan your days ahead and use the program to keep track of your tasks!

> If u fail to plan, you plan to fail!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

quzhetao01 committed:
Added printLine function

quzhetao01 committed:
Completed skeletal version Level-0

quzhetao01 committed:
Added bye control flow logic

quzhetao01 committed:
Added list feature

quzhetao01 committed:
Finish completeTask functionality

quzhetao01 committed:
Added ToDos class

quzhetao01 committed:
Added Deadline class

quzhetao01 committed:
Add Event class

quzhetao01 committed:
Finish integrating different tasks tracking

quzhetao01 committed:
Added TextUiTesting

quzhetao01 committed:
Add error handling

quzhetao01 committed:
Added error handling for mark and unmark

quzhetao01 committed:
Remove unused code

quzhetao01 committed:
Added delete functionality

quzhetao01 committed:
Add writing functionality

quzhetao01 committed:
Add file reading functionality

quzhetao01 committed:
Merge branch 'branch-Level-7'

quzhetao01 committed:
Add LocalDate functionality

quzhetao01 committed:
Merge branch 'branch-Level-8'

quzhetao01 committed:
Add more OOP

quzhetao01 committed:
Merge branch 'master' into add-gradle-support

quzhetao01 committed:
Add gradle build

quzhetao01 committed:
Fix gradle config

quzhetao01 committed:
Merge branch 'add-gradle-support'

quzhetao01 committed:
Add Test Cases for Storage and Parser

quzhetao01 committed:
Add javadocs for Ui class

quzhetao01 committed:
Add javadoc for all classes

quzhetao01 committed:
Add Find Task functionality

quzhetao01 committed:
Fix code style

quzhetao01 committed:
Merge branch 'branch-A-CodingStandard'

quzhetao01 committed:
Merge branch 'branch-A-JavaDoc'

quzhetao01 committed:
Merge branch 'branch-Level-9'

tanyyyming commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    /**
+     * Main method to start the Duke application.
```
I am unsure whether this Javadoc comment should start with a verb instead. Maybe something like `Starts the Duke application`?

quzhetao01 replied:
Hi thanks for pointing that out! Have resolved it!

tanyyyming commented on src/main/java/duke/command/AddEventCommand.java:
```diff
@@ -0,0 +1,44 @@
+	 */
+	@Override
+	public void execute(FunnyList taskList, Ui ui, Storage storage) throws DukeException  {
+		Event task = new Event(this.taskDescription, false, this.from, this.to);
```
Maybe not use the `this` keyword since the fields `from` and `to` are not shadowed in this method?
```suggestion
		Event task = new Event(this.taskDescription, false, from, to);
```
The same suggestion applies to several other places.

tanyyyming commented on src/main/java/duke/list/FunnyList.java:
```diff
@@ -0,0 +1,99 @@
+        ArrayList<Task> result = new ArrayList<>();
+        for (int i = 0; i < this.taskList.size(); i++) {
+            Task curr = this.taskList.get(i);
+            if (curr.match(search)) {
```
Good compliance with the coding standard here!
> Single statement conditionals should still be wrapped by curly brackets. 
―Java Coding Standard

tanyyyming commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,85 @@
+			ArrayList<Task> taskList = new ArrayList<>();
+			while (s.hasNext()) {
+				String[] row  = s.nextLine().split("\\|");
+				switch (row.length) {
```
The `switch-case` statement needs to follow the coding standard (no indentation for the `case` clause).
Maybe also check your tab size to make sure its equal to four spaces? The indentation here seems a bit off to me.

quzhetao01 replied:
oh i see, i must have missed this out. Thank you!

tanyyyming commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,264 @@
+public class ParserTest {
+
+	@Test
+	public void success_AddTodo() {
```
Same problem as mentioned before. The tab size seems a bit off. Maybe double check to ensure your tab size is consistent throughout the project files?

quzhetao01 replied:
thanks for the pointer hmm i am also a little confused as it appears fine on my IDE but not on github, i will look into this problem, thanks!

tanyyyming reviewed (commented):
In general, well done! 

The code is clean and the majority follows the Java coding standard. There are only a few violations that I've marked out.

However, maybe due to some tab size setting problem, the indentations in the code base are a bit inconsistent. I believe they can be fixed in the future easily.

quzhetao01 committed:
Fix indentation issues

quzhetao01 committed:
Add step1-3 of tutorial

quzhetao01 committed:
Add MainWindow controller

quzhetao01 committed:
Implement GUI

quzhetao01 committed:
Merge branch 'branch-Level-10'

tim-pipi commented on src/main/java/duke/command/Command.java:
Nice use of abstract class `Command`. Solid job!

tim-pipi commented on src/main/java/duke/list/FunnyList.java:
```diff
@@ -0,0 +1,99 @@
+     * @return The task that was undone.
+     * @throws DukeException If the index is out of bounds.
+     */
+    public Task undoTask(int index) throws DukeException {
```
Perhaps `unmarkTask` would be a better name here?

tim-pipi commented on src/main/java/duke/list/FunnyList.java:
```diff
@@ -0,0 +1,99 @@
+     * @return The completed task.
+     * @throws DukeException If the index is out of bounds.
+     */
+    public Task completeTask(int index) throws DukeException {
```
Perhaps `markTask` would be a better name?

tim-pipi commented on src/main/java/duke/list/FunnyList.java:
```diff
@@ -0,0 +1,99 @@
+ * This class manages a list of tasks and provides methods for adding, retrieving,
+ * completing, undoing, and deleting tasks.
+ */
+public class FunnyList {
```
I didn't quite like the naming of this class 😞 It may be confusing to the other readers. Perhaps `TaskList` would be a better name?

quzhetao01 replied:
okay i will do something about this thank you!

tim-pipi commented on src/main/java/duke/ui/Ui.java:
I really like the way you handle everything about user interactions in the `Ui` class! 👍

tim-pipi reviewed (commented):
LGTM. Just a few nits to pick about Naming!

quzhetao01 committed:
Fix coding standard violation and naming

quzhetao01 committed:
Merge branch 'fix-coding-standard'

AlfredBeNoel commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,99 @@
+ * task processing, and exception handling.
+ */
+public class Duke {
+    private Storage storage;
```
Overall I think your code looks good, just some few nits here and there, other than that good job! 😄

AlfredBeNoel commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * A GUI for Duke using FXML.
+ */
+public class Main extends Application {
```
Great job in creating a separate file just for the Main.java class!

AlfredBeNoel commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,35 @@
+ * Represents a command in the Duke application.
+ * This abstract class serves as the base for different types of commands that can be executed.
+ */
+public abstract class Command {
```
Great work in classifying all the different commands using a separate command class!

AlfredBeNoel commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @return The completed task.
+     */
+    public Task completeTask() {
```
Perhaps a better name for this method would be `doTask()` ? So that it may comply with the `undoTask()` method

quzhetao01 committed:
Update build.gradle file

The main class in the build.gradle file is referring to the duke.java

Implementation of the duke application now relies on JavaFX.
duke.Launcher is the file responsible for launching the GUI and the
Duke application as a whole. Running the current JAR file does not
cause the GUI the pop up.

Let's change the main class in the build.gradle file to duke.Launcher.

The JAR file created after this change will reflect the GUI to
appear, which represents the current version of the duke
application.

Some unused code was removed as well in this commit.

quzhetao01 committed:
Add assertions to Deadline and Event

quzhetao01 committed:
Fix Code Quality of Duke

There are unused imports and unused code some files. The parse method in
Parser.java is very long.

Unused imports and code will confuse the readers. Long code such as the one
in Parser.java is very hard to understand for future developers as well.

Let's remove unused imports and code. Let's also abstract the code in each
case statement in the parse method to its own separate function.

Doing so will make the code much more readable and understandable.
Abstracting the logic in each case statement will shorten the code
in the parse method.

quzhetao01 committed:
Merge pull request #1 from quzhetao01/branch-A-Assertions

Add assertions to Deadline and Event

quzhetao01 committed:
Merge updated master code

quzhetao01 committed:
Merge pull request #2 from quzhetao01/branch-A-CodeQuality

Fix Code Quality of Duke

quzhetao01 committed:
Add CI/CD

quzhetao01 committed:
Add reschedule feature

quzhetao01 committed:
Merge branch 'branch-B-Snooze'

quzhetao01 committed:
Add window title to ui

quzhetao01 committed:
Update ReadME
